### ISSUE TITLE
When a file has syntax errors, the project gives strange errors from a terminal

### ISSUE THREAD

**Issue by tavurth** [godotengine/godot#73358]
### Godot version

v4.0.beta14.official [28a24639c]

### System information

Mac

### Issue description

When I have a syntax error such as `mixing tabs and spaces` in any given file.

When I then run the project from a terminal instead of the `godot` editor.

I see strange errors:

`SCRIPT ERROR: Parse Error: Could not preload resource script "res://addons/cs-some/something.gd".`

I must then open this file in the Godot editor to figure out exactly on which line the error occurs.

### Steps to reproduce

1. Add an extra `:` somewhere in any file
2. Try to run your project from the terminal

### Minimal reproduction project

See above

**Comment by akien-mga** [comment]
That's normal, the scene you're currently editing doesn't use this script directly, but preloads something which depends on this script (or preloads the script itself). preloading fails because the script is invalid.

**Comment by tavurth** [comment]
Hmm, would it not be more useful to have the error shown in the terminal?

I would say that 80% of the time I don't have the Godot editor open, and this forces me to load the project, find the file and investigate the error.

I was so happy when Godot started to fully support external editors, but this feels like a small regression 🤔

**Comment by Zireael07** [comment]
> Hmm, would it not be more useful to have the error shown in the terminal?

Agreed, it would be nice to have this IF you're using an external editor

**Comment by tavurth** [comment]
Rather more importantly. When a file has syntax errors when running from the editor, the file is not debugged.

<img width="404" alt="Screenshot 2023-03-03 at 21 50 17" src="https://user-images.githubusercontent.com/100964/222751186-47c97d02-7ee3-4ce4-a3d7-99acc06385c3.png">

<img width="489" alt="Screenshot 2023-03-03 at 21 49 53" src="https://user-images.githubusercontent.com/100964/222751077-6f9bffd5-8a60-4296-bd77-5ce019267412.png">

I must first open the `Map.gd` file to find the error:

<img width="732" alt="Screenshot 2023-03-03 at 21 50 38" src="https://user-images.githubusercontent.com/100964/222751255-94e145b4-fee7-464c-ba20-fb20b1f6e86a.png">

v4.0.stable.official [92bee43ad]

**Comment by tavurth** [comment]
@akien-mga is that also expected behaviour?

**Comment by akien-mga** [comment]
This looks fine to me. The error tells you that the script can't be preloaded. Opening it lets you investigate why.

**Comment by tavurth** [comment]
It's not only preload though, `load` gives the same behaviour.

<img width="429" alt="Screenshot 2023-03-04 at 19 32 35" src="https://user-images.githubusercontent.com/100964/222901251-6b84d837-eecb-4bd2-813d-b359475bc60b.png">

<img width="773" alt="Screenshot 2023-03-04 at 19 32 41" src="https://user-images.githubusercontent.com/100964/222901262-8dbc0e5d-7967-47fd-81b0-bc4d4708d3ef.png">

<img width="883" alt="Screenshot 2023-03-04 at 19 32 49" src="https://user-images.githubusercontent.com/100964/222901268-760757a2-a6fb-4227-ab4b-4a6fdc6f42d5.png">

IMO this is a bit confusing for newer users. It would be a much nicer flow to open the file to the offending line, instead of the file which is trying to `load` or `preload` that file.

**Comment by tavurth** [comment]
This bug currently makes `Godot 4` very tricky to work with. 

I've included a screenshot:

<img width="772" alt="Screenshot 2023-04-09 at 21 31 29" src="https://user-images.githubusercontent.com/100964/230778820-01f392ed-43c6-419c-b835-7d1dd7a104df.png">

So while doing some refactoring, I changed some things, and now everything is broken.

In `Character.gd` I can only see a reference to another bug:
<img width="657" alt="Screenshot 2023-04-09 at 21 32 59" src="https://user-images.githubusercontent.com/100964/230778878-1c115a87-027c-4893-b922-78c6b62fb319.png">


And in `SignedResource.gd` I can only see another reference to somewhere else:
<img width="402" alt="Screenshot 2023-04-09 at 21 33 31" src="https://user-images.githubusercontent.com/100964/230778905-d719b476-f58e-47b4-8219-55ca79d2c84f.png">


And the final bug?
<img width="517" alt="Screenshot 2023-04-09 at 21 34 10" src="https://user-images.githubusercontent.com/100964/230778942-bc04391c-74c8-4e1e-8648-a689b416a91b.png">


I've no idea how to go about even thinking to fix this. However I do hope that someone notices this soon 🙏

**Comment by tavurth** [comment]
@akien-mga can you take another look at this with the updated bug report please? 🙏

**Comment by DarkMessiah** [comment]
![image](https://user-images.githubusercontent.com/1756388/233180271-c0bbf868-6d00-4dca-8ecd-c92dba0506ee.png)

You need commas here :)

**Comment by tavurth** [comment]
@DarkMessiah I think you did not read the issue closely? 🤔

**Comment by HolonProduction** [comment]
The editor does display errors in dependent scripts by now:

<img width="521" height="167" alt="Image" src="https://github.com/user-attachments/assets/7f7f4f7f-b66c-4183-919f-09c5b6507cb5" />

The console is also displaying script errors that occur while loading:

<img width="728" height="164" alt="Image" src="https://github.com/user-attachments/assets/14b90d70-fe9a-4334-9bae-4e52bf4f26b5" />
